packages/react-router/src/router.ts

```diff
@@ -1180,6 +1199,41 @@ export class Router<
       // And also update the searchError if there is one
       match.searchError = searchError
 
+      const parentMatchId = parentMatch?.id
+
+      const parentContext = !parentMatchId
+        ? ((this.options.context as any) ?? {})
+        : (parentMatch.context ?? this.options.context ?? {})
+
+      match.context = {
+        ...parentContext,
+        ...match.__routeContext,
+        ...match.__beforeLoadContext,
+      }
+
+      // Update the match's context
+      const contextFnContext: RouteContextOptions<any, any, any, any> = {
+        search: match.search,
+        params: match.params,
+        context: match.context,
+        location: next,
+        navigate: (opts: any) =>
+          this.navigate({ ...opts, _fromLocation: next }),
+        buildLocation: this.buildLocation,
+        cause: match.cause,
```

Comment:
@tannerlinsley not sure if this was intentionally omitted or not, but `cause`, `abortController`, and `preload`, were not present on this object before but were _typed_ on the `RouteContextOptions` type.

Reply:
Yeah, I made it the same as `beforeLoad` in this regard. If the types are wrong here they can be removed

Reply:
could we take this chance to add previous and next location (in parsed form) as params to all of those functions?

Reply:
@schiller-manuel I'd push for that kind of change to go separate from this PR.

A change of this nature would require more sweeping changes to make sure those values are present in the `loadMatches` function, whilst not really contributing to the task of adding this `context` function to the route definition and wiring it up.

Once this goes in, if we are adding in the _previous_ and _next_ location options into the shared definition used for `context`, `beforeLoad`, and the `loader`, it'd certainly be more straight-forward.